phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java

```diff
@@ -166,9 +168,14 @@ public static void assertTaskColumns(Connection conn, String expectedStatus, PTa
         String taskStatus = rs.getString(PhoenixDatabaseMetaData.TASK_STATUS);
         assertEquals(expectedStatus, taskStatus);
 
+        if (expectedTableName != null) {
+            String tableName = rs.getString(PhoenixDatabaseMetaData.TABLE_NAME);
+            assertEquals(expectedTableName, tableName);
+        }
+
         if (expectedData != null) {
             String data = rs.getString(PhoenixDatabaseMetaData.TASK_DATA);
-            assertEquals(expectedData, data);
+            assertTrue(data.contains(expectedData));
```

Comment:
I ran `DropTableWithViewsIT` and it looks like `data` just gets the value `{"TaskDetails":"COMPLETED"}`, whereas when running `IndexMetadataIT`, it gets a json such as `{"IndexName":"R_ASYNCIND_N000002","RebuildAll":true,"JobID":"job_local1895022878_0001","TaskDetails":"COMPLETED"}`. 

To make this check more accurate, can we pass in a key and expected value and parse the data accordingly? Just checking `contains` may cause us to miss actual failures or even lead to false alarms if we change values stored in the `TASK_DATA`.

Reply:
Key is a bit complicated:
PRIMARY KEY (" + TASK_TYPE + "," + TASK_TS + " ROW_TIMESTAMP," + TENANT_ID + "," + TABLE_SCHEM + "," +
            TABLE_NAME + "))
I will try to move to key

Reply:
I meant, key="IndexName" and value="<name to find" which we then parse from the `TASK_DATA` string, rather than just doing a `contains` within the entire string. I wasn't talking about the primary key, sorry for the confusion.